docs/blender_docs/io_gltf2.rst

```diff
@@ -30,7 +30,198 @@ This importer/exporter supports the following glTF 2.0 features:
+and should be converted to meshes prior to export.
+
+Materials
+---------
+
+The core material system in glTF supports a metal/rough PBR workflow with the following
+channels of information:
+
+- Base Color
+- Metallic
+- Roughness
+- Baked Ambient Occlusion
+- Normal Map
+- Emissive
+
+.. note::
+
+   When images are used to drive these channels, glTF requires that PNG or JPG format
+   is used to pack the images.
+
+.. note::
+
+   glTF does not directly contain Blender's nodes, however, this glTF exporter
+   and importer will use nodes to describe the contents of the glTF file.
+
+Base Color
+^^^^^^^^^^
+
+The glTF base color is determined by looking for a Base Color input on a
+"Principled BSDF" node.  If the input is disconnected, the input's default color
+(the color swatch next to the disconnected input) is used as the Base Color for
+the material.
+
+If an "Image Texture" node is found to be connected to the Base Color input, that
+image will be used as the glTF base color.
+
+Metallic and Roughness
+^^^^^^^^^^^^^^^^^^^^^^
+
+These values are read from the "Principled BSDF" node.  If both of these inputs
+are disconnected, the node will display sliders to control their respective
+values between 0.0 and 1.0, and these values will be copied into the glTF.
+
+When using an image, glTF expects the metallic values to be encoded in the
+Blue (``B``) channel, and roughness to be encoded in the Green (``G``) channel of the
+same image.  If images are connected to the Blender node in a manner that
+does not follow this convention, this addon may attempt to adapt the image
+to the correct form during export, costing some performance.
+
+In the Blender node graph, it is recommended to use a "Separate RGB" node
+to separate the channels from an "Image Texture" node, and connect the
+Green (``G``) channel to Roughness, and Blue (``B``) to Metallic.  The glTF exporter
+will recognize this arrangement as matching the glTF standard, and that will
+allow it to simply copy the image texture into the glTF file during export.
+
+The Image Texture node for this should have its "Colorspace" setting
+configured to "Non-Color Data".
+
+.. figure:: /images/addons_io-gltf2-material-metalRough.png
+   :alt: A metallic-roughness image connected in a manner consistent
+         with the glTF standard, allowing it to be used verbatim inside
+         an exported glTF file.
+
+   A metallic/roughness image connected in a manner consistent
+   with the glTF standard, allowing it to be used verbatim inside
+   an exported glTF file.
+
+Baked Ambient Occlusion
+^^^^^^^^^^^^^^^^^^^^^^^
+
+glTF is capable of storing a baked ambient occlusion map.  Currently there
+is no arrangement of nodes that causes Blender to use such a map in exactly
+the same way as glTF intends it to be used.  However, if the exporter finds
+a custom node group by the name of ``glTF Metallic Roughness``, and finds an
```

Comment:
I don't actually know how to create a custom node group, other than importing one from the old exporter.. this guidance seems a bit unclear without that.

Arguably it would be nice if we detected an Image Texture named `Occlusion`, regardless of  what it's connected to.

Reply:
Yes.  I was thinking we still have to so something better about Occlusion, but I couldn't write documentation on what that might be until after it gets implemented.  So the PR for that should include an update to this section, or a new issue to update this section.

Creating custom node groups must be covered elsewhere in the Blender manual.  It seemed inappropriate to delve into those details here.

Reply:
As a point of reference, I knew nothing of how to create Node Groups but a quick search found the answer (CTRL+G in the Node Editor). However, following the instructions for adding an 'Occlusion' input didn't cause any AO information to be exported. I tend to agree with the suggestion of simply having an Image Texture named `Occlusion` without the need for a Node Group, although it does keep things tidier.

Image shows the Node Group in the foreground, main graph in background with collapsed Node Group on the far left.
![image](https://user-images.githubusercontent.com/31968496/52757586-053fca80-2fba-11e9-82ce-e583b3c81d3a.png)

Reply:
@j-conrad Creating custom node groups seems a bit tricky (at least in 2.80), since it doesn't seem possible to create an empty group when no other node groups are selected. I found that I needed to create some input, select it, press `Ctrl + G` in the Shader Editor and then customize it in the properties panel (press `N`).

That said, I believe the node graph above is incorrect; the texture shouldn't be part of the group at all and should just be passed to an input named `Occlusion`. It's also important that the node group is named `glTF Metallic Roughness`.

This worked for me, for both double-sided materials and baked occlusion:

![image](https://user-images.githubusercontent.com/4733018/52767372-4951c000-2ff8-11e9-8b77-b046df6ee679.png)

Reply:
One thing I'm unclear on, however, is whether or not `Separate RGB` is needed as shown below:

![image](https://user-images.githubusercontent.com/4733018/52769443-4ce84580-2ffe-11e9-8723-5dc26927d2d3.png)

The result appeared the same when not using a `Metallic` or `Roughness` map, but I did notice that without `Separate RGB`, the texture was monochromatic black and white, and monochromatic cyan when splitting off the red channel.

No `Separate RGB`:

![image](https://user-images.githubusercontent.com/4733018/52769988-f11ebc00-2fff-11e9-9225-c0fee864be21.png)

`Separate RGB` using R channel:


![image](https://user-images.githubusercontent.com/4733018/52769725-1d860880-2fff-11e9-96a1-6c707df17aba.png)

When combining with `Metallic` (`B`) and `Roughness` (`G`), I can only assume that the separation is necessary, but would appreciate any clarification on this.

Reply:
I would think your selection of ‘non-color data’ would be enough to not require splitting out the red channel, but I’m new enough to Blender to not know for sure.

I think what confuses me is that we’re putting Occlusion in a separate image texture to begin with. Why can’t it live within the Red channel of the Metallic/Roughness image? I understand that the Principled BSDF node doesn’t have an AO input (which is why we’re having to do any of this in the first place). But it seems like the exporter could simply look for data in that Red channel and assume it’s the Ambient Occlusion map.

Reply:
It’s worth noting that it’s a fairly common practice to already have your Ambient Occlusion stores there (Red channel). Common enough that you see people referring to the texture as “ARM”

Reply:
"ORM" please, not "ARM" 😸 

But yes, the exporter was never made to understand the "Separate RGB" node being used with occlusion, even though glTF stores occlusion in Red.  This deficiency is just a quirk of the history of the project, not a real design decision.

Ideally, there should be a single ORM image connected to a single Separate RGB node, and G and B should be connected to Principled BSDF, and the R side should be connected to... something.  What the "something" could be, is still up for debate.  We need to figure it out, implement it, and then document whatever it is.

Reply:
oops..  Yeah, I definitely meant 'ORM'.

Using the Node Group is working for me now too, including 'DoubleSided'. Although, I noticed that naming the input `doubleSided` (camel case, as per the glTF spec) does not work. Seems better to use the glTF spec names verbatim, unless there's some special reason not to.

Also, what do you think about the idea of using the Node Group as the single place to put any additional inputs, for example, `baseColorFactor`, `metallicFactor` and `roughnessFactor`? I understand that those two values automatically appear in the JSON when a texture is not specified, but since the spec states that it's possible to use a texture AND have it multiplied by a Factor, it seems like a good place to specify in such a scenario since right now, if a texture gets used, those factors disappear.

Reply:
Actually the factors work by creating extra nodes that multiply with the Image Texture nodes.  I mention this in the PR here, but I don't have a screenshot for it.  I think if you have such a glTF already, the importer may generate the correct pattern of nodes for this.

The input names, and for that matter the entire behavior of the `glTF Metallic Roughness` custom node, is legacy.  They date from the previous exporter which worked exclusively off a custom node grouping, and was not compatible with Principled BSDF.  The names are meant to be user-facing, so they do not follow a camelCase pattern.

Reply:
Regarding camelCase, I suppose that makes sense. Moving forward, most users will be using this new Blender documentation for reference without needing to look at the glTF specs. In the interim, for people like me, it has the potential to cause some mistakes but I can deal with it.

I had completely forgotten about the glTF **importer**-- what a great way to understand how a glTF is set up by seeing nodes get generated.  :)

Reply:
@j-conrad If any clarification is still needed for using a single ORM texture, I believe you can use something like the graph below:

![image](https://user-images.githubusercontent.com/4733018/52835512-b0cd4580-30b4-11e9-8581-6d1d04149285.png)

Please let me know if I'm mistaken.

Reply:
That’s exactly what my graph looks like and is also working. Thanks!